src/middleware/secure-headers/index.ts

```diff
@@ -129,15 +171,49 @@ function getFilteredHeaders(options: SecureHeadersOptions): [string, string][] {
 
 function getCSPDirectives(
   contentSecurityPolicy: SecureHeadersOptions['contentSecurityPolicy']
-): string {
-  return Object.entries(contentSecurityPolicy || [])
-    .map(([directive, value]) => {
-      const kebabCaseDirective = directive.replace(/[A-Z]+(?![a-z])|[A-Z]/g, (match, offset) =>
-        offset ? '-' + match.toLowerCase() : match.toLowerCase()
-      )
-      return `${kebabCaseDirective} ${Array.isArray(value) ? value.join(' ') : value}`
+): [SecureHeadersCallback | undefined, string | string[]] {
+  const callbacks: ((ctx: Context, values: string[]) => void)[] = []
+  const resultValues: string[] = []
+
+  for (const [directive, value] of Object.entries(contentSecurityPolicy as any)) {
```

Comment:
This casting is not necessary if `contentSecurityPolicy` is `ContentSecurityPolicyOptions`.

Reply:
fixed in [a6e1277](https://github.com/honojs/hono/pull/2577/commits/a6e1277e949deaf6aebe1fa516a14644aa69dbab)